unable to install it on osx

Hi there,

I know this project is only tested on linux but as everything should work on OSX, I gave it a try. Unfortunately it did not compile and people on [users.rust.com](https://users.rust-lang.org/t/cannot-compile-libunicorn-sys-on-osx/9050) were unable to help me so before I dig deep into the rust code, I wanted to know whether you can image what the problem is. So here is what I did:

1. I add`unicorn = "0.4.1"` as dependency
2. I run `cargo run --verbose` but I get the following error:
```
$ cargo run --verbose
       Fresh libc v0.2.20
       Fresh bitflags v0.6.0
       Fresh gcc v0.3.41
       Fresh byteorder v1.0.0
   Compiling libunicorn-sys v0.4.0
     Running `rustc /Users/$username/.cargo/registry/src/github.com-1ecc6299db9ec823/libunicorn-sys-0.4.0/lib.rs --crate-name libunicorn_sys \
--crate-type lib -g -C metadata=271778d4aa553e9c -C extra-filename=-271778d4aa553e9c --out-dir \
/Users/$username/dev/$project/target/debug/deps --emit=dep-info,link -L \
dependency=/Users/$username/dev/$project/target/debug/deps --extern \
libc=/Users/$username/dev/$project/target/debug/deps/liblibc-29ef97a68464c2b7.rlib --extern \
bitflags=/Users/$username/dev/$project/target/debug/deps/libbitflags-b8c7fb7df9b2bc2e.rlib --cap-lints allow -L \
native=/Users/$username/dev/$project/target/debug/build/libunicorn-sys-fe33b86568c508cb/out -l static=unicorn`

error: failed to add native library /Users/$user/dev/$project/target/debug/build/libunicorn-sys-fe33b86568c508cb/out/libunicorn.a: failed to open archive

error: Could not compile `libunicorn-sys`.

Caused by:
  process didn't exit successfully: `rustc /Users/$user/.cargo/registry/src/github.com-1ecc6299db9ec823/libunicorn-sys-0.4.0/lib.rs --crate-name libunicorn_sys --crate-type lib -g -C metadata=271778d4aa553e9c -C extra-filename=-271778d4aa553e9c --out-dir /Users/$user/dev/$project/target/debug/deps --emit=dep-info,link -L dependency=/Users/$user/dev/$project/target/debug/deps --extern libc=/Users/$user/dev/$project/target/debug/deps/liblibc-29ef97a68464c2b7.rlib --extern bitflags=/Users/$user/dev/$project/target/debug/deps/libbitflags-b8c7fb7df9b2bc2e.rlib --cap-lints allow -L native=/Users/$user/dev/$project/target/debug/build/libunicorn-sys-fe33b86568c508cb/out -l static=unicorn` (exit code: 101)
```
3. The file `libunicorn.a` exists and file reveals the following:
```
$ file /Users/$username/dev/$project/target/debug/build/libunicorn-sys-fe33b86568c508cb/out/libunicorn.a
/Users/$username/dev/$project/target/debug/build/libunicorn-sys-fe33b86568c508cb/out/libunicorn.a: Mach-O universal binary with 2 architectures: [x86_64: current ar archive random library] [i386]
/Users/$username/dev/$project/target/debug/build/libunicorn-sys-fe33b86568c508cb/out/libunicorn.a (for architecture x86_64):	current ar archive random library
/Users/$username/dev/$project/target/debug/build/libunicorn-sys-fe33b86568c508cb/out/libunicorn.a (for architecture i386):	current ar archive random library
```
  I assume the error gets thrown [here](https://github.com/rust-lang/rust/blob/master/src/librustc_trans/back/archive.rs#L192) but if possible I would like to avoid digging there. `RUST_BACKTRACE=1` does not reveal more information.
4. my versions:
```
$ rustc --version
rustc 1.14.0
$ cargo --version
cargo 0.15.0-dev (298a012 2016-12-20)
```

By looking at your files, it does not look like that you are doing any linux specific things, but maybe I did not see it. One thing I suspect is that rustc expects a linux `.a` and not an osx `.a`, but according to the documentation, rustc should also work on osx. I am happy to fix this bug and file a merge request afterwards, but I do not know where to start.

```
$ cargo test
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling gcc v0.3.41
   Compiling bitflags v0.6.0
   Compiling libc v0.2.20
   Compiling libunicorn-sys v0.4.0 (file:///Users/jonasfelber/Downloads/unicorn-rs/libunicorn-sys)
   Compiling unicorn v0.4.1 (file:///Users/jonasfelber/Downloads/unicorn-rs)
error[E0252]: a type named `uc_handle` has already been imported in this module
  --> src/lib.rs:48:15
   |
37 | use ffi::*;
   |     ------- previous import of `uc_handle` here
...
48 | pub use ffi::{uc_handle, uc_hook};
   |               ^^^^^^^^^ already imported

error[E0252]: a type named `uc_hook` has already been imported in this module
  --> src/lib.rs:48:26
   |
37 | use ffi::*;
   |     ------- previous import of `uc_hook` here
...
48 | pub use ffi::{uc_handle, uc_hook};
   |                          ^^^^^^^ already imported

error[E0252]: a module named `unicorn_const` has already been imported in this module
  --> src/lib.rs:49:9
   |
37 | use ffi::*;
   |     ------- previous import of `unicorn_const` here
...
49 | pub use ffi::unicorn_const as unicorn_const;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ already imported

error[E0252]: a type named `uc_handle` has already been imported in this module
  --> src/lib.rs:48:15
   |
37 | use ffi::*;
   |     ------- previous import of `uc_handle` here
...
48 | pub use ffi::{uc_handle, uc_hook};
   |               ^^^^^^^^^ already imported

error[E0252]: a type named `uc_hook` has already been imported in this module
  --> src/lib.rs:48:26
   |
37 | use ffi::*;
   |     ------- previous import of `uc_hook` here
...
48 | pub use ffi::{uc_handle, uc_hook};
   |                          ^^^^^^^ already imported

error[E0252]: a module named `unicorn_const` has already been imported in this module
  --> src/lib.rs:49:9
   |
37 | use ffi::*;
   |     ------- previous import of `unicorn_const` here
...
49 | pub use ffi::unicorn_const as unicorn_const;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ already imported

error: aborting due to 3 previous errors

Build failed, waiting for other jobs to finish...
error: aborting due to 3 previous errors

error: Could not compile `unicorn`.

To learn more, run the command again with --verbose.
```
Does not work.

--- a/libunicorn-sys/Cargo.toml
+++ b/libunicorn-sys/Cargo.toml
@@ -18,6 +18,7 @@ name = "libunicorn_sys"
 
 [build-dependencies]
 gcc = "0.3"
+os_type="0.5.0"
 
 [dependencies]
 bitflags = "0.6"

--- a/libunicorn-sys/build.rs
+++ b/libunicorn-sys/build.rs
@@ -10,7 +10,7 @@ fn main() {
             .status();
     }
     let out_dir = env::var("OUT_DIR").unwrap();
-    let _ = Command::new("./make.sh").current_dir("unicorn").status();
+    let _ = Command::new("./make.sh").args(&["macos-universal-no"]).current_dir("unicorn").status();
     let unicorn = "libunicorn.a";
     let _ = Command::new("cp").current_dir("unicorn").arg(&unicorn).arg(&out_dir).status();
     println!("cargo:rustc-link-search=native={}", out_dir);

--- a/libunicorn-sys/build.rs
+++ b/libunicorn-sys/build.rs
@@ -1,16 +1,26 @@
 extern crate gcc;
+extern crate os_type;
 
-use std::path::{Path};
+use std::path::Path;
 use std::process::Command;
 use std::env;
 
 fn main() {
     if !Path::new("unicorn/.git").exists() {
-        let _ = Command::new("git").args(&["submodule", "update", "--init", "--depth", "5"])
+        let _ = Command::new("git")
+            .args(&["submodule", "update", "--init", "--depth", "5"])
             .status();
     }
     let out_dir = env::var("OUT_DIR").unwrap();
-    let _ = Command::new("./make.sh").args(&["macos-universal-no"]).current_dir("unicorn").status();
+
+    let make_args = match os_type::current_platform() {
+        os_type::OSType::OSX => ["macos-universal-no"],
+        _ => [""],
+    };
+
+    // TODO(sduquette): the package build should fail if this command fails.
+    let _ = Command::new("./make.sh").args(&make_args).current_dir("unicorn").status();
+
     let unicorn = "libunicorn.a";
     let _ = Command::new("cp").current_dir("unicorn").arg(&unicorn).arg(&out_dir).status();
     println!("cargo:rustc-link-search=native={}", out_dir);

